lib/BoundaryValueDiffEqFIRK/src/sparse_jacobians.jl

```diff
@@ -12,27 +12,34 @@ If the problem is a TwoPointBVProblem, then this is the complete Jacobian, else
 computes the sparse part excluding the contributions from the boundary conditions.
 """
 function __generate_sparse_jacobian_prototype(
-        ::FIRKCacheNested, ::StandardBVProblem, ya, yb, M, N)
+        ::FIRKCacheNested, ::StandardBVProblem, ya, yb, M, N, ad)
     fast_scalar_indexing(ya) ||
         error("Sparse Jacobians are only supported for Fast Scalar Index-able Arrays")
     J_c = BandedMatrix(Ones{eltype(ya)}(M * (N - 1), M * N), (1, 2M - 1))
-    return ColoredMatrix(J_c, matrix_colors(J_c'), matrix_colors(J_c))
+    problem = ColoringProblem(;
+        partition = ifelse((ADTypes.mode(ad) isa ADTypes.ReverseMode), :row, :column))
+    algo = GreedyColoringAlgorithm()
```

Comment:
At some spots in the code you use `LargestFirst()` while at others you do not, is it on purpose?